### DISCUSSION TITLE
Failed to set fetch Cache on vercel build

### DISCUSSION THREAD

**Discussion by MasteringBOX** [vercel/community#4290] [Bug Report]
### Page to Investigate

Fails on Vercel Build

### Steps to Reproduce

1. Create a fetch function with default cache and call your API.
2. Call such funcion in a page.
3. Deploy to vercel.
4. Check Server logs (use your own API for that).

I get this error on my vercel build. I can build the project in my local enviroment without issue.

If I set fetch to "no-cache" the error goes away but the performance is horrible obviously. I tried also using React cache as an alternative but it doesn't seem cache anything. 

```
Failed to set fetch cache https://www.example.com/wp-api/wp/v2/posts?per_page=50&page=1 TypeError: fetch failed
--
14:02:30.463 | at Object.fetch (node:internal/deps/undici/undici:11576:11)
14:02:30.463 | at async globalThis.fetch (/vercel/path0/node_modules/next/dist/server/lib/patch-fetch.js:199:16)
14:02:30.463 | at async invokeRequest (/vercel/path0/node_modules/next/dist/server/lib/server-ipc/invoke-request.js:17:12)
14:02:30.463 | at async invokeIpcMethod (/vercel/path0/node_modules/next/dist/server/lib/server-ipc/request-utils.js:45:21)
14:02:30.464 | at async /vercel/path0/node_modules/next/dist/server/lib/patch-fetch.js:396:29 {
14:02:30.464 | cause: Error: write ECONNRESET
14:02:30.464 | at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:94:16)
14:02:30.464 | at WriteWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
14:02:30.464 | errno: -104,
14:02:30.464 | code: 'ECONNRESET',
14:02:30.464 | syscall: 'write'
14:02:30.464 | }
```
Any help appreciated, if you need any other info please ask. lt seems a Vercel issue though.

EDIT: I created to projects showcasing the issue. The problem is that in build fetch cache misses and the API gets called a lot. In local this gets solved downgrading to nextjs 13.4.3. This solves the issue partially in Vercel, although I still get some duplicated API calls that should not happen. 

Simple project to reproduce:
https://github.com/MasteringBOX/fetch-bug
A bit more complex project with more fetches:
https://github.com/MasteringBOX/fetch-cache-bug

I will post this in nextjs as well.

**Comment by amyegan** [comment]
Hey @MasteringBOX. Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that we can use to help with debugging?

**Comment by MasteringBOX** [comment]
Hi @amyegan,

Not at the moment because its a private repository and I can't share this. But I can create a small project that reproduces the issue. I'll update this when its on github.

Regards

Edit: The issue is with the nextjs version 13.5.3, fetch is not caching anything and for every single page there's a new API request. I was able to reproduce this locally. Downgrading to 13.4.3 fixes the issue locally but vercel still calls the API on every single page. I edited my post with the reproducible examples.
